### ISSUE TITLE
Unexpected type mismatch with alias types.

### ISSUE THREAD

**Issue by vsaulue** [Benjamin-Dobell/IntelliJ-Luanalysis#85]
# Environment

   name   |   version  
----------|----------
 IDEA version         | 2021.1.1 Community Edition (build #IC-211.7142.45)
 Luanalysis version | 1.2.3
 OS                        | Windows 10

# What are the steps to reproduce this issue?

Create a Lua file with the following content:

```
---@alias set<x> table<x,any>

---@generic T
---@param value set<T>
---@return set<T>
local function identity(value)
    return value
end

---@type set<set<number>>
local aSet = {}

---@type set<set<number>>
local bSet = identity(aSet)
```

## _Maintainer Edit: Minimal Reproduction_

```lua
---@alias set<x> table<x,any>

---@type set<set<number>>
local aSet

aSet = aSet
```

# What happens?

Luanalysis generates a "Type mismatch" error on line `local bSet = identity(aSet)`:

```
Type mismatch. Required: 'set<set<number>>' Found: 'set<table<number, any>>'
```

![image](https://user-images.githubusercontent.com/34475300/122940521-d483b300-d374-11eb-9971-5ff39f1bc669.png)

# What were you expecting to happen?

No error: this should be a valid assign.

# Any logs, error output, etc?

None

# Any other comments?

The problem seems to only happen with `alias` types involved. The following lines do not generate an error

```
---@type set<table<number,any>>
local cSet = identity(aSet)
---@type table<table<number,any>,any>>
local cSet = identity(aSet)
```
